Methods journal recommendation (#7)
greenelab/annorxiver_manuscript: content/03.methods.md

For each principal component we calculated its cosine similarity with  all tokens in our word2vec model's vocabulary.
We report the top 100 positive and negative scoring tokens in the form of  word clouds, where the size of each word corresponds to the magnitude of similarity and color represents positive (blue) or negative (orange) association.

### Journal Recommendation

> We trained two k-nearest neighbor classifiers to recommend journals based on article content.

Reviewer: ```suggestion
We sought to predict the journal that a paper would eventually be published in based on its embeddings.
We examined predictions against both individual papers as well as journal centroids using k-nearest neighbor classifiers.
```
I wanted to give just a little bit more intro here.

> The first classifier recommends journals based on articles in close proximity to the query and the second classifier recommends journals based on close proximity to each journal's centroid.

Reviewer: ```suggestion
The first classifier recommended journals based on the articles in close proximity to the query article and the second classifier recommended journals based on the journal centroids in close proximity to the query article.
```
What do you think of this phrasing? I switched it to past tense & tried to make "query article" consistent. I think it's not quite right and needs further adjustment though.

We used PubMed Central's (PMC) open access dataset to train both models.

> We preprocessed this dataset by reserving articles that were found in both PMC and bioRxiv for our test set and used the rest for training.

Reviewer: ```suggestion
A subset of bioRxiv preprints were directly linked to PMC papers because they had been published.
We held all of these out as our test set.
We used the remainder of the PMC corpus for training.
We embedded each PMC paper into the embeddings trained on bioRxiv preprints.
```

> Furhermore, we filtered out journals that had less than 100 articles within PMC and downsampled the remaining journals to only have 100 articles as this prevents overrepresentation of mega journals.

Reviewer: ```suggestion
Furhermore, we filtered out journals that had fewer than 100 articles within PMC and downsampled the remaining journals to only have 100 articles, which prevented mega journals from completely dominating the predictions.
```

Reviewer: Did you do this just for the article-based predictor? I thought you were going to calculate centroids with all papers.

> We compared two distance metrics (Eucledian distance and Manhattan distance) for both classifiers.

Reviewer: Unless you're going to present results/comparison for both, the performance was _so similar_ for these that I'd probably just recommend going with the one you're using and ignoring this comparison in the paper.

> We report our results in the form of accuracy; however, we considered a prediction to be positive if the correct journal appeared within the ten closest neighbors of a query.

Reviewer: ```suggestion
Our goal was to present the recommendations on a website as a short list.
We considered a list of roughly ten potential journals to be appropriate and considered a prediction to be correct if the correct journal appeared within the ten closest neighbors of a query.
For the paper-based classifier, because multiple papers may come from the same journal this may return fewer than ten potential matches.
```

> The results of the first classifier were generated using 10-fold cross validation on the downsampled training set.

Reviewer: I wonder if it would make sense to have what you have at the very top to explain the two classifiers, then to fully explain the article-based classifier, then the journal-based one. There are some subtle differences in here that might come out more effectively that way.

For each principal component we calculated its cosine similarity with  all tokens in our word2vec model's vocabulary.
We report the top 100 positive and negative scoring tokens in the form of  word clouds, where the size of each word corresponds to the magnitude of similarity and color represents positive (blue) or negative (orange) association.

### Journal Recommendation

We aimed to predict the journal a paper would eventually be published in based on its embedding representation.
We illustrate our recommendations as a short list along with a network visualization available at [website link here once available](link goes here when live).

> Since our approach is solely focused on embeddings, we utilized the k-nearest neighbors technique with Euclidean distance to recommend journals.

Reviewer: ```suggestion
Since we sought to examine if embeddings were related to publication venue, we used a simple k-nearest neighbors approach with Euclidean distance to recommend journals.
```

First, we filtered all journals that had fewer than 100 papers in the PMC dataset.
A subset of our PMC corpus was directly linked to papers in bioRxiv as they had been published as open access articles.
We held out this subset and treated it as our gold standard test set.

> The remainder of the PMC corpus was used for training and initial evaluation via cross validation.

Reviewer: ```suggestion
We used the remainder of the PMC corpus for training and initial evaluation via cross validation.
```

> We considered a list of roughly ten potential journals to be feasible for broadcasting to the user and we treated a prediction to be correct if the correct journal appeared within the ten closest neighbors of the query article.

Reviewer: ```suggestion
We considered a list of ten journal suggestions to be an appropriate number and we considered a prediction to be a true positive if the correct journal appeared within the ten closest neighbors of the query article.
```

> Certain journals publish articles whose topics are remarkably niche, while others publish articles that cover a broad array of topics.

Reviewer: ```suggestion
Certain journals publish articles in a focused topic are, while others publish articles that cover many topics.
```

Conversely, predictions using this technique could be biased due to the overrepresentation of general topic journals.
We call this approach the paper-based classifier.
This classifier takes a query article that has been projected onto the embedding space trained on bioRxiv preprints as input and reports the journals of the top ten closest papers.
The number of journals returned via this method could be less than ten as multiple papers in close proximity to query article may belong to the same journal. 

The second approach is based on close proximity to a journal's centroid.
This technique provides recommendations that are more focused on domain-specific publication venues.
We call this approach the journal-based classifier.
This classifier was trained by computing journal centroids via taking the average embedding of all papers published in each journal.
Following the centroid calculation, this classifier takes a query article projected onto the same embedding space as above for input and reports the top ten nearest journals centroids.
Both the paper-based classifier and the journal-based classifier were optimized via 10-fold cross validation.

> Lastly, we evaluated performance of both classifiers on our golden standard test set.

Reviewer: ```suggestion
Lastly, we evaluated performance of both classifiers on our gold standard test set of published preprints.
```
